controllers/disruption_schedule_controller.go

```diff
@@ -0,0 +1,221 @@
+func (r *DisruptionScheduleReconciler) getScheduledTimeForDisruption(disruption *chaosv1beta1.Disruption) (*time.Time, error) {
+	timeRaw := disruption.Annotations[ScheduledAtAnnotation]
+	if len(timeRaw) == 0 {
+		return nil, nil
+	}
+
+	timeParsed, err := time.Parse(time.RFC3339, timeRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	return &timeParsed, nil
+}
+
+func (r *DisruptionScheduleReconciler) checkTargetResourceExists(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) (bool, error) {
+	var targetObj client.Object
+
+	switch instance.Spec.TargetResource.Kind {
+	case "Deployment":
+		targetObj = &appsv1.Deployment{}
+	case "StatefulSet":
+		targetObj = &appsv1.StatefulSet{}
+	}
+
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      instance.Spec.TargetResource.Name,
+		Namespace: instance.Namespace,
+	}, targetObj)
+
+	return errors.IsNotFound(err), err
+}
+
+func (r *DisruptionScheduleReconciler) updateTargetResourcePreviouslyMissing(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) (bool, bool, error) {
+	disruptionScheduleDeleted := false
+	targetResourceNotFound, err := r.checkTargetResourceExists(ctx, instance)
+
+	if targetResourceNotFound {
+		r.log.Warnw("target does not exist, this schedule will be deleted if that continues", "error", err)
+
+		if instance.Status.TargetResourcePreviouslyMissing == nil {
+			r.log.Warnw("target is missing for the first time, updating status", "targetPreviouslyMissing", instance.Status.TargetResourcePreviouslyMissing)
+
+			return targetResourceNotFound, disruptionScheduleDeleted, r.handleTargetResourceFirstMissing(ctx, instance)
+		}
+
+		if time.Since(instance.Status.TargetResourcePreviouslyMissing.Time) > time.Hour*24 {
+			r.log.Errorw("target has been missing for over one day, deleting this schedule",
+				"timeMissing", time.Since(instance.Status.TargetResourcePreviouslyMissing.Time))
+
+			disruptionScheduleDeleted = true
+
+			return targetResourceNotFound, disruptionScheduleDeleted, r.handleTargetResourceMissingOverOneDay(ctx, instance)
+		}
+	} else if instance.Status.TargetResourcePreviouslyMissing != nil {
+		r.log.Infow("target was previously missing, but now present. updating the status accordingly")
+
+		return targetResourceNotFound, disruptionScheduleDeleted, r.handleTargetResourceNowPresent(ctx, instance)
+	}
+
+	return targetResourceNotFound, disruptionScheduleDeleted, nil
+}
+
+func (r *DisruptionScheduleReconciler) handleTargetResourceFirstMissing(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) error {
+	instance.Status.TargetResourcePreviouslyMissing = &metav1.Time{Time: time.Now()}
+	if err := r.Client.Status().Update(ctx, instance); err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	return nil
+}
+
+func (r *DisruptionScheduleReconciler) handleTargetResourceMissingOverOneDay(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) error {
```

Comment:
```suggestion
func (r *DisruptionScheduleReconciler) handleTargetResourceMissingOverExpiration(ctx context.Context, instance *chaosv1beta1.DisruptionSchedule) error {
```

No need to couple the function name to a value we'll likely make configurable soon enough

Reply:
`TooLong`, `PastExpiration`, etc are all valid